### PR TITLE
Mock-wallet-data

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: legacy/crypto/walletManager.js [hunk 1]
```diff
@@ -154,6 +154,14 @@ class WalletManager {
     return this._wallets
   }
 
+  getWallet() {
```
Comment (stackchain) [inline legacy/crypto/walletManager.js:157]:
```suggestion
  get wallet() {
```

File: legacy/crypto/walletManager.js [hunk 2]
```diff
@@ -222,7 +230,7 @@ class WalletManager {
    */
 
   get isInitialized() {
-    return this._wallet
+    return this.getWallet().isInitialized
```
Comment (stackchain) [inline legacy/crypto/walletManager.js:233]:
```suggestion
    return this.wallet.isInitialized
```

File: legacy/crypto/walletManager.js (hunk 1 above)
Comment (stackchain) [reply legacy/crypto/walletManager.js:157]: my concern is only about the `throw` here, throwing inside getters would not be a good practice

**Comment by stackchain** [comment]
I think we should consider implementing the state pattern for the wallet.

File: legacy/crypto/walletManager.js (hunk 1 above)
Comment (wolverineks) [reply legacy/crypto/walletManager.js:157]: why make it a getter?

**Comment by wolverineks** [comment]
> I think we should consider implementing the state pattern for the wallet.

I'm not sure what you mean. could you elaborate?

**Review by stackchain — APPROVED** [review]

**Pull request merged**